ext/mysql2/extconf.rb

```diff
@@ -94,18 +94,32 @@ def asplode lib
 end
 
 # This is our wishlist. We use whichever flags work on the host.
-# TODO: fix statement.c and remove -Wno-declaration-after-statement
-# TODO: fix gperf mysql_enc_name_to_ruby.h and remove -Wno-missing-field-initializers
-%w(
-  -Wall
-  -Wextra
-  -Werror
-  -Wno-unused-function
-  -Wno-declaration-after-statement
-  -Wno-missing-field-initializers
-).select do |flag|
-  try_link('int main() {return 0;}', flag)
-end.each do |flag|
+# -Wall and -Wextra are included by default.
+[
+  '-fsanitize=address',
+  '-fsanitize=cfi',
+  '-fsanitize=integer',
+  '-fsanitize=memory',
+  '-fsanitize=thread',
+  '-fsanitize=undefined',
+  '-Werror',
+  '-Weverything',
```

Comment:
I'm concerned that -Weverything is an open-ended list, so future compilers will add in more checks that we might not pass. I would use -Weverything without -Werror, but given the number of allowances below, I don't want to use -Weverything.